### ISSUE TITLE
How to compare the accuracies?

### ISSUE THREAD

**Issue by pratibha77118** [ubsuny/23-Homework6G4#34]
In task 1 it is said to compare the accuracies (how many digits are correct) and efficiencies (how many steps does it take to reach a given accuracy). How are we supposed to take the accuracy value. Should we take adaptive trapezoid method as reference or any approximate value of true integral to compare the accuracy?

**Comment by tirthbha** [comment]
@pratibha77118  I have sent this group a merge request as integral accuracy, hope that file (integral_accuracy.py)  will answer your question.

**Comment by s4il3sh** [comment]
> In task 1 it is said to compare the accuracies (how many digits are correct) and efficiencies (how many steps does it take to reach a given accuracy). How are we supposed to take the accuracy value. Should we take adaptive trapezoid method as reference or any approximate value of true integral to compare the accuracy?

My question exactly.

> @pratibha77118 I have sent this group a merge request as integral accuracy, hope that file (integral_accuracy.py) will answer your question. #37

Thanks.

**Comment by pratibha77118** [comment]
@tirthbha  I did check on that but I got this warning while implementing scipy 
<ipython-input-19-f02a56b956f4>:28: IntegrationWarning: The maximum number of subdivisions (50) has been achieved.
  If increasing the limit yields no improvement it is advised to analyze 
  the integrand in order to determine the difficulties.  If the position of a 
  local difficulty can be determined (singularity, discontinuity) one will 
  probably gain from splitting up the interval and calling the integrator 
  on the subranges.  Perhaps a special-purpose integrator should be used.
  result, error = quad(integrand, a, b)
[21]

**Comment by tirthbha** [comment]
@pratibha77118 Please add n to the last line as "  return h * s, n ", I forgot to add n there. I will work.

**Comment by tirthbha** [comment]
> > In task 1 it is said to compare the accuracies (how many digits are correct) and efficiencies (how many steps does it take to reach a given accuracy). How are we supposed to take the accuracy value. Should we take adaptive trapezoid method as reference or any approximate value of true integral to compare the accuracy?
> 
> My question exactly.
> 
> > @pratibha77118 I have sent this group a merge request as integral accuracy, hope that file (integral_accuracy.py) will answer your question. #37
> 
> Thanks. So you have used the exact value obtained from scipy.

Yes, that is taken as a reference value to compare with others obtained from different methods.

**Comment by tirthbha** [comment]
![Screenshot from 2023-11-19 21-42-54](https://github.com/ubsuny/23-Homework6G4/assets/143649367/1740b496-fbd3-436a-b574-0dc635c7b31a)
I got result like this.

**Comment by tirthbha** [comment]
> > In task 1 it is said to compare the accuracies (how many digits are correct) and efficiencies (how many steps does it take to reach a given accuracy). How are we supposed to take the accuracy value. Should we take adaptive trapezoid method as reference or any approximate value of true integral to compare the accuracy?
> 
> My question exactly.
> 
> > @pratibha77118 I have sent this group a merge request as integral accuracy, hope that file (integral_accuracy.py) will answer your question. #37
> 
> Thanks. So you have used the exact value obtained from scipy.

Yes

**Comment by Mosaddeq107** [comment]
> > > In task 1 it is said to compare the accuracies (how many digits are correct) and efficiencies (how many steps does it take to reach a given accuracy). How are we supposed to take the accuracy value. Should we take adaptive trapezoid method as reference or any approximate value of true integral to compare the accuracy?
> > 
> > 
> > My question exactly.
> > > @pratibha77118 I have sent this group a merge request as integral accuracy, hope that file (integral_accuracy.py) will answer your question. #37
> > 
> > 
> > Thanks. So you have used the exact value obtained from scipy.
> 
> Yes

Your request has been approved.
